src/tree/branch.rs

```diff
@@ -42,33 +45,45 @@ impl AsRef<[BlsScalar]> for PoseidonLevel {
 }
 
 /// Represents a full path for a merkle opening
-#[derive(Debug, Clone, Copy)]
-pub struct PoseidonBranch<const DEPTH: usize>([PoseidonLevel; DEPTH]);
+#[derive(Debug, Clone)]
+pub struct PoseidonBranch<const DEPTH: usize> {
+    path: Vec<PoseidonLevel>,
+}
```

Comment:
I guess that const generics just constraint the type here to disseminate between `PoseidonBranch`es of different depths. Is that the case?

Reply:
Also, I see that you didn't go for the Boxed array and the root attrb. At the end it;s the same. But would be nice to add some docs on how the leafs/levels are positioned in the `PoseidonBranch`

Reply:
The idea was to mirror the implementation we have for kelvin so we have a consistent allocator across the stack.
https://github.com/dusk-network/microkelvin/blob/main/src/branch.rs#L88

The `root` attribute can be created, but to remove it from the path would require some unnecessary refactoring. I believe it makes more sense to have an iterator over the path that just ends up with the root. By separating one from the other, we will need to increase the code complexity a lot.

The `Vec` solves the unstable `DEPTH + 1` approach so we don't really have anything to restrict this implementation

Reply:
Makes sense. If the rest are OK keeping the const-generics to constraint the type, for me it's ok.